x/ibc/light-clients/solomachine/types/proof.go

```diff
@@ -34,182 +34,281 @@ func EvidenceSignBytes(sequence uint64, data []byte) []byte {
 }
 
 // HeaderSignBytes returns the sign bytes for verification of misbehaviour.
-//
-// Format: {sequence}{header.newPubKey}
-func HeaderSignBytes(header *Header) []byte {
-	return append(
-		sdk.Uint64ToBigEndian(header.Sequence),
-		header.GetPubKey().Bytes()...,
-	)
+func HeaderSignBytes(
+	cdc codec.BinaryMarshaler,
+	header *Header,
+) ([]byte, error) {
+	data := &HeaderData{
+		NewPubKey:      header.NewPublicKey,
+		NewDiversifier: header.NewDiversifier,
+	}
+
+	dataBz, err := cdc.MarshalBinaryBare(data)
```

Comment:
I don't know why I didn't think of this before, but we could make all of these `Any`s and make `SignBytes` and interface with the function `GetSequence`

Reply:
this is probably the cleaner approach? thoughts?

Reply:
I don't understand what the proposed approach is

Reply:
```go
type SignBytesI interface {
    GetSequence() uint64
}

registry.RegisterImplementations(
    (*exported.SignBytes)(nil),
       &HeaderSignBytes{},
       &ClientStateSignBytes{},
       etc.
)

// a marshalled any == sign bytes
signBytes, err := MarshalClientStateSignBytes(any)

// unmarshal any to interface to call GetSequence for misbehaviour
var signBytes SignBytesI
codec.UnmarshalAny(msg.Signature.Data, signBytes)
signBytes.GetSequence()
```

Every specific SignBytes definition implements SignBytes. Each is marshalled to/from as an`Any` before being signed/verified. That way when we handle evidence we unmarshal to `SignBytes` interface and call `GetSequence`. 

I didn't lookup the exact naming/structure some of my pseudo code might be off.

Reply:
maybe this is too much, but the basic gist of the problem is given some arbitrary `[]byte` we need to get the sequence from it in misbehaviour